### DISCUSSION TITLE
Velociraptor Communication matrix

### DISCUSSION THREAD

**Discussion by mtabousha** [Velocidex/velociraptor#2927] [Q&A]
Hey Velocigurus!
i am in phase to open the communication between VR server/clients.
Need to know the bidirectional communication matrix to allow it from Firewall side.

**Comment by scudette** [comment]
I am not sure what you mean by Communication Matrix but the communication protocol is described here 
https://docs.velociraptor.app/blog/2020/2020-09-28-velociraptor-network-communications-30568624043a/
https://docs.velociraptor.app/docs/deployment/troubleshooting/#velociraptor-communications

Generally you only need to open the port on which the clients are connecting - for self signed ssl deployments you can set the port in the config file but the default is 8889 and for lets encrypt deployments it is 443

As of 0.7.0 you can add source IP filtering to the GUI handlers so it is ok to expose them both on port 443 (which the lets encrypt deployment does) 
https://docs.velociraptor.app/docs/deployment/references/#GUI.allowed_cidr
